File: src/prog_langs/java/read.java
import java.sql.*;
import java.util.*;

class ReadDB {
    // 取得できるようにstatic finalで変数を格納しておく
    static final String URL  = "jdbc:mysql://localhost/java";
    static final String USER = "guest";
    static final String PASS = "DK4XMDVATL2N8448CMF9";
        
    public static void main(String[] args) {
        // 実行する文
        String sql = "SELECT * FROM toy;";
        
        try ( Connection connection = DriverManager.getConnection(URL, USER, PASS);
              PreparedStatement statement = connection.prepareStatement(sql); )
        {
            // 実行結果を取得する
            ResultSet result = statement.executeQuery();
            // 配列に格納するパターンを実行するために、配列と2次元配列を準備しておく
            ArrayList<String> res_arr = new ArrayList<>();
            Map<Integer, String> res_map = new HashMap<Integer, String>();

            System.out.println("whileで出力");
            // while文で回す
            while (result.next()) {
                // 単純に出力
                System.out.println(result.getString("id")+ " : " +result.getString("name"));
                // 配列に格納するパターン
                res_arr.add(result.getString("id")+ " : " +result.getString("name"));
                // 2次元配列に格納するパターン
                res_map.put(result.getInt("id"), result.getString("name"));
            }

            System.out.println("forEachで出力");
            res_arr.forEach(
                // 配列を出力
                value -> System.out.println(value)
            );

            System.out.println("多次元を出力");
            res_map.forEach(
                // 多次元配列を出力
                (key, value) -> System.out.println(key + " : " + value)
            );

        } catch (SQLException e) {
            e.printStackTrace();
        }
    }
}

File: src/prog_langs/go/read.go
package main

import (
    "log"
    "fmt"
    "database/sql"
    _ "github.com/go-sql-driver/mysql"
)

// selectしたデータを格納するToy構造体を宣言しておく
type Toy struct {
    id   int
    name string
}

func main() {
    // データベースのハンドルを取得する
    db, err := sql.Open("mysql", "guest:DK4XMDVATL2N8448CMF9@tcp(localhost)/go")
    if err != nil {
      // エラーが発生したらログを残す
      log.Fatal(err)
    }
    // selectを実行してデータ取得
    rows, err := db.Query("select id, name from toy")
    // toyResult=Toy構造体
    var toyResult []Toy

    // selectで取得したデータで配列を作成
    for rows.Next() {
        // toyをToy構造体にする
        var toy Toy
        // エラーが発生したらログを残す
        if err := rows.Scan(&toy.id,&toy.name); err != nil {
            log.Fatal(err)
        }
        // データを配列に格納
        toyResult = append(toyResult,toy)
    }

    // 作成した配列を出力
    for _,u := range toyResult{
        fmt.Println("id: ", u.id,", name: ", u.name)
    }
    
    // close
    defer db.Close()
}
